R/seroprevalence_data.R

```diff
@@ -148,7 +174,12 @@ prepare_bin_data <- function(serodata) {
 #'   \item{`tsur`}{Year of the survey}
 #' }
 #' @param foi Numeric atomic vector corresponding to the desired
-#'   Force-of-Infection ordered from past to present
+#' time-varying or age-varying Force-of-Infection to simulate from
+#' @param mu Seroreversion rate
+#' @param is_time_varying Boolean specifying whether `foi`
+#' is a time-varying function
+#' @param is_age_varying Boolean specifying whether `foi`
+#' is an age-varying function
```

Comment:
I wonder: would these be better off as a single parameter, `model_type = c('age', 'time', 'age-time')`? I think that would resolve the ambiguity you have to handle and easily allow generalisation to other types of FOIs.